content/source/docs/enterprise/api/variables.html.md

```diff
@@ -13,25 +13,23 @@ This set of APIs covers create, update, list and delete operations on variables.
 
 ## Create a Variable
 
-| Method | Path           |
-| :----- | :------------- |
-| POST | /vars |
-
-### JSON API Attributes
-
-This POST endpoint requires a JSON payload, which uses the following keys:
-
-- `data`
-    - `type` (`string: <required>`) - Must be `"vars"`.
-    - `attributes`
-        - `key` (`string: <required>`) - specifies the name of the variable which will be passed into the plan/apply.
-        - `value` (`string: <required>`) - specifies the value of the variable which will be passed into the plan/apply.
-        - `category` (`string: "terraform"|"env"`) - specifies whether this should be parsed as a Terraform variable (with support for HCL) or as an environment variable. This governs how it is accessible in the Terraform configuration.
-        - `hcl` (`bool: false`) - use HCL when setting the value of the string.
-        - `sensitive` (`bool: false`) - marks the variable as sensitive. If true then the variable is written once and not visible thereafter.
-- `filter`
-    - `workspace.name` (`string: required`) - variables must be associated with a workspace. Specify the workspace's name with the `filter` query parameter.
-    - `organization.name` (`string: required`) - workspaces must be owned by an organization. Specify which organization owns the workspace with the `filter` query parameter.
+`POST /vars`
+
+### Request Body
+
+This POST endpoint requires a JSON object with the following properties as a request payload. (Properties without a default value are required.)
+
+Key path                    | Type   | Default | Description
+----------------------------|--------|---------|------------
+`data.type`                 | string |         | Must be `"vars"`.
+`data.attributes.key`       | string |         | The name of the variable.
+`data.attributes.value`     | string |         | The value of the variable.
+`data.attributes.category`  | string |         | Whether this is a Terraform or environment variable. Valid values are `"terraform"` or `"env"`.
+`data.attributes.hcl`       | bool   | `false` | Whether to evaluate the value of the variable as a string of HCL code. Has no effect for environment variables.
+`data.attributes.sensitive` | bool   | `false` | Whether the value is sensitive. If true then the variable is written once and not visible thereafter.
+`filter.workspace.name`     | string |         | The name of the workspace that owns the variable.
```

Comment:
Can we move this out of the JSON API payload and keep it as part of the query parameters? Even if we require code changes I think a single way of filtering would be preferable to consumers of the API.

Reply:
It looks like the JSON API supports `filter` so we're going to allow it in both circumstances. Slightly confusing, but 🤷‍♂️.